engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -824,7 +824,7 @@ private DiskProfile allocateTemplatedVolume(Type type, String name, DiskOffering
         int volumesNumber = 1;
         List<DatadiskTO> templateAsIsDisks = null;
         String configurationId = null;
-        if (template.isDeployAsIs()) {
+        if (template.isDeployAsIs() && vm.getType() != VirtualMachine.Type.SecondaryStorageVm) {
```

Comment:
do we have to also keep a check for other system VMs ? or pass this condition only for user VMs ? I see a problem when CPVM is tried to recreate when SSVM is not there, we might get into same issue.

Reply:
Good point; @nvazquez can you check for CPVM and other systemvm types as well?

Reply:
I've pushed a fix to check for running SSVMs in the zone as well

Reply:
Thanks for fixing that case @nvazquez